variants/tamc_termod_s3/variant.cpp

```diff
@@ -15,24 +15,10 @@ float getBatteryCapacity() {
 }
 
 bool getChargingState() {
-  pinMode(CHG, INPUT_PULLUP);
   return !digitalRead(CHG);
 }
 
-void (*__onChargeStart__)();
-void (*__onChargeEnd__)();
-void setOnChargeStart(void (*func)()) { __onChargeStart__ = func; }
-void setOnChargeEnd(void (*func)()) { __onChargeEnd__ = func; }
-
-void ARDUINO_ISR_ATTR chargeIsr() {
-  if (getChargingState()) {
-    __onChargeStart__();
-  } else {
-    __onChargeEnd__();
-  }
-}
-
 extern "C" void initVariant(void){
-  attachInterrupt(CHG, chargeIsr, CHANGE);
+  pinMode(CHG, INPUT_PULLUP);
```

Comment:
what would happen if you keep everything original, but you remove `pinMode(CHG, INPUT_PULLUP);` from `getChargingState()` and put it before `attachInterrupt(CHG, chargeIsr, CHANGE);` in `initVariant(void)`?

Reply:
Yes! It worked! Thank you. It keeps rebooting before and i didn't know why. And ISR is not that important, so i just remove it. Thanks a lot! Will commit another pull request.